src/saml.ts

```diff
@@ -146,6 +148,7 @@ class SAML {
       path: ctorOptions.path ?? "/saml/consume",
       host: ctorOptions.host ?? "localhost",
       issuer: ctorOptions.issuer ?? "onelogin_saml",
+      audience: ctorOptions.audience ?? ctorOptions.issuer ?? "unknown_audience", // use issuer as default
```

Comment:
This would be much clearer if `issuer` would be modified to required option (i.e. user must provide non null string instead of defaulting `null/undefined` to `onelogin_saml`). 

If someone set `ctor.issuer = null` and doesn't set `audience` or sets `audience = null` result would be
```javascript
...
issuer: "onelogin_saml",
audience: "unknown_audience",
...
```
which is not what documentation says ("if audience is not provided it deafults to issuer").

Reply:
I agree, and I had planned on a follow-up PR to address this. It appears that the cascading changes are making the reviewing harder, and I didn't want to add to it.